src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -1513,8 +1524,20 @@ public DependencyGraph(Core core)
             graphNodeMap = new Dictionary<ulong, List<GraphNode>>();
         }
 
+        private List<GraphNode> cachedNodes;
+
         public List<GraphNode> GetGraphNodesAtScope(int classIndex, int procIndex)
         {
+            if (classIndex == Constants.kInvalidPC && procIndex == Constants.kInvalidPC)
+            {
+                if (cachedNodes == null)
+                {
+                    graphNodeMap.TryGetValue(GetGraphNodeKey(classIndex, procIndex), out cachedNodes);
+                }
+
+                return cachedNodes;
+            }
+
```

Comment:
This is trying to create and escape for the lookup the most common lookup if for the default kInvalidPC pair.  Maybe a better way to do that or document that